kanidmd/src/lib/actors/v1_write.rs

```diff
@@ -624,6 +625,26 @@ impl QueryServerWriteV1 {
                             .remove_account_webauthn(&mut audit, &rwe)
                             .and_then(|r| idms_prox_write.commit(&mut audit).map(|_| r))
                     }
+                    SetCredentialRequest::GenerateBackupCode => {
+                        let gbe = GenerateBackupCodeEvent::from_parts(
+                            &mut audit,
+                            // &idms_prox_write.qs_write,
+                            ident,
+                            target_uuid,
+                        )
+                        .map_err(|e| {
+                            ladmin_error!(
+                                audit,
+                                "Failed to begin internal_credential_set_message: {:?}",
+                                e
+                            );
+                            e
+                        })?;
+                        idms_prox_write
+                            .generate_backup_code(&mut audit, &gbe)
+                            .and_then(|r| idms_prox_write.commit(&mut audit).map(|_| r))
+                            .map(SetCredentialResponse::Token)
```

Comment:
This should probably use a new type like BackupCodes on the `SetCredentialResponse` enum so it's really clear what these are.